flow/Makefile.py

```diff
@@ -18,159 +19,39 @@ def main():
 
     # Parse values out of provided "config.mk" file
     config = parse_config_mk.parse(args['DESIGN_CONFIG'])
-    design = config['DESIGN_NAME']
-
-    # Perform variable substitution where required.
-    repls = {}
-    # Set DESIGN_NICKNAME = DESIGN if not already set.
-    if not 'DESIGN_NICKNAME' in config:
-        repls['DESIGN_NICKNAME'] = design
-    # Set PLATFORM_DIR if not already set.
-    if not 'PLATFORM_DIR' in config:
-        repls['PLATFORM_DIR'] = os.path.abspath(os.path.join(mydir, 'platforms', config['PLATFORM']))
-    if not 'CORNER' in config:
-        if config['PLATFORM'] == 'asap7':
-            repls['CORNER'] = 'BC'
-        else:
-            repls['CORNER'] = 'typical'
-
-    # Setup timing corner libraries if supported for preprocessing. TODO: Kind of hack-y.
-    if config['PLATFORM'] == 'asap7':
-        corners = {'BC': 'FF', 'WC': 'SS', 'TC': 'TT'}
-        post = 'nldm_201020.lib'
-        for corner, abrv in corners.items():
-            os.environ[f'{corner}_DFF_LIB_FILE'] = os.path.abspath(os.path.join('platforms', 'asap7', 'lib', f'asap7sc7p5t_SEQ_RVT_{abrv}_{post}'))
-
-    if repls:
-        # Set default platform values.
-        # Set replacement value in os.environ so that the Makefile parser uses it.
-        for k, v in repls.items():
-            os.environ[k] = v
-        # Re-parse the Makefile
-        config = parse_config_mk.parse(args['DESIGN_CONFIG'])
-        # Update 'config' dictionary with the substitutions used to generate it.
-        for k, v in repls.items():
-            config[k] = v
+    # We'll use 'design' for finding the config.mk file in the target setup method, and
+    # there is inconsistency between whether that corresponds to 'DESIGN_NAME' or 'DESIGN_NICKNAME'.
+    # So, set design name based on file path.
+    design_name_match = re.search(f'\/[a-zA-Z0-9_]+\/config.mk', args['DESIGN_CONFIG']).group(0)
+    design = design_name_match[1:-len('/config.mk')]
```

Comment:
Nit - could simplify a bit by using a capture group in the regex (add parens around the part you want to extract): 

```suggestion
    design = re.search(f'\/([a-zA-Z0-9_]+)\/config.mk', args['DESIGN_CONFIG']).group(1)
```